src/frontend/komponenter/Felleskomponenter/HeaderMedSøk/HeaderMedSøk.tsx

```diff
@@ -0,0 +1,77 @@
+import AlertStripe from 'nav-frontend-alertstriper';
+import React from 'react';
+import { useHistory } from 'react-router';
+
+import { Header, Søk } from '@navikt/familie-header';
+
+import { søkFagsaker } from '../../../api/fagsak';
+import { IFagsakDeltager } from '../../../typer/fagsakdeltager';
+import { Ressurs, RessursStatus } from '../../../typer/ressurs';
+import FagsakDeltagerskort from './FagsakDeltagerskort';
+import validator from '@navikt/fnrvalidator';
+
+export interface IHeaderMedSøkProps {
+    brukerNavn?: string;
+    brukerEnhet?: string;
+}
+
+export const HeaderMedSøk: React.FunctionComponent<IHeaderMedSøkProps> = ({ brukerNavn, brukerEnhet }) => {
+    const [resultat, settResultat] = React.useState<IFagsakDeltager[] | undefined>(undefined);
+    const [spinner, settSpinner] = React.useState<boolean>(false);
+    const [søkfeil, settSøkfeil] = React.useState<string | undefined>(undefined);
+
+    const history = useHistory();
+
+    const søk = (v: string) => {
+        slettResultat();
+        settSpinner(true)
+        søkFagsaker(v).then((response: Ressurs<IFagsakDeltager[]>) => {
+            settSpinner(false);
+            if (response.status === RessursStatus.SUKSESS) {
+                settSøkfeil(undefined);
+                settResultat(response.data);
+            } else if (response.status === RessursStatus.IKKE_HENTET) {
+                settSøkfeil("Person ikke funnet");
+            } else if (response.status === RessursStatus.FEILET || response.status === RessursStatus.IKKE_TILGANG) {
+                settSøkfeil(response.melding);
+            }
+        }).catch(error => {
+            settSøkfeil("Ukjent API feil: " + error);
+        });
+    }
+
+    const slettResultat = () => {
+        settSøkfeil(undefined);
+        settResultat(undefined);
+    }
+
+    const korter = resultat && (resultat.length > 0 ? resultat.map((deltager, index) => {
```

Comment:
What does this do? Why not include it in the return directly?
If you want to seperate the logic I suggest creating a new component.

Also it is very hard to understand what is going on here. Many different outcomes and multiple elvis operators. Rewrite this to something which is easier to read.

Reply:
It is to display the search results or alert stripe, depending on the search status.
I thought it is easier to read than a couple of "if", but ok, I can write it in another way if you like.

Reply:
In this case I would probably move the logic to a separate component(s) with good names to indicate what is happening.